Shared/Samples/Geocode offline/GeocodeOfflineView.Model.swift

```diff
@@ -0,0 +1,206 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+extension GeocodeOfflineView {
+    /// The view model for the sample.
+    @MainActor
+    class Model: ObservableObject {
+        // MARK: Properties
+        
+        /// A map with a tiled layer of the streets in San Diego, CA, USA.
+        let map = {
+            let map = Map()
+            
+            // Create a tiled layer using the local tile package.
+            let tileCache = TileCache(fileURL: .streetMap)
+            let tiledLayer = ArcGISTiledLayer(tileCache: tileCache)
+            
+            // Add the layer to the map.
+            map.addOperationalLayer(tiledLayer)
+            
+            return map
+        }()
+        
+        /// The graphics overlay for the marker graphic.
+        let graphicsOverlay = GraphicsOverlay()
+        
+        /// The red map marker graphic used to indicate a given location on the map.
+        private let markerGraphic = {
+            // Create a symbol using the image from the project assets.
+            guard let markerImage = UIImage(named: "RedMarker") else { return Graphic() }
+            let markerSymbol = PictureMarkerSymbol(image: markerImage)
+            
+            // Change the symbol's offsets, so it aligns properly to a given point.
+            markerSymbol.leaderOffsetY = markerImage.size.height / 2
+            markerSymbol.offsetY = markerImage.size.height / 2
+            
+            // Create a graphic with the symbol.
+            return Graphic(symbol: markerSymbol)
+        }()
+    
+        /// The value used to offset the callout when the map magnifier is showing.
+        private let magnifierOffset = {
+            // Get the magnifier image from ArcGIS package bundle.
+            let bundle = Bundle(identifier: "com.esri.ArcGIS") ?? Bundle(identifier: "ArcGIS")
+            let image = UIImage(named: "Magnifier", in: bundle, with: nil) ?? UIImage()
+            
+            // Determine the amount by which we will need to offset the callout along y-axis.
+            return CGPoint(x: .zero, y: -image.size.height)
+        }()
+        
+        /// The locator task used to preform the geocode operations, loaded from a local file.
+        private let locatorTask = LocatorTask(name: "SanDiego_StreetAddress", bundle: .main)
+        
+        /// The viewpoint of the map view, initially centered on San Diego, CA, USA.
+        @Published var viewpoint = Viewpoint(
+            center: Point(x: -13_042_250, y: 3_857_970, spatialReference: .webMercator),
+            scale: 2e4
+        )
+        
+        /// The placement of the callout on the map.
+        @Published var calloutPlacement: CalloutPlacement?
```

Comment:
It would be ideal to keep the `calloutPlacement` a property of the view.

Reply:
It might not be easy to do, but still feel like the callout placement should be in the view rather than the model.

1. Move the `calloutPlacement`, `calloutShouldOffset` and `calloutText` into the view
2. Move `updateCalloutPlacement` to the view
3. Since the change of `searchText` affects `calloutPlacement`, it can trigger the child view to update
4. The `geocodeSearch` and `reverseGeocode` methods can both return a `GeocodeResult?`, and let the view unpack the result to get the attributes and `displayLocation`

Reply:
Letting the the views unpack the `GeocodeResult?` would require both the views to have access to the callout variable since the `geocodeSearch` is called from `GeocodeOfflineView` and the `reverseGeocode` is called from `GeocodeMapView`. Are you suggesting to put the callout variables on the `GeocodeOfflineView` and then pass them all to the `GeocodeMapView`?  Thoughts @rolson?